pkg/applier/inventory/inventory_converter_test.go

```diff
@@ -0,0 +1,647 @@
+							Kind:      "obj2-kind",
+						},
+						Strategy:   actuation.ActuationStrategyApply,
+						Actuation:  actuation.ActuationPending,
+						Reconcile:  actuation.ReconcilePending,
+						UID:        "obj2-uid",
+						Generation: int64(12),
+					},
+				},
+			}.build(t),
+			wantObj: resourceGroupFactory{
+				spec: v1alpha1.ResourceGroupSpec{
+					Resources: []v1alpha1.ObjMetadata{
+						{
+							Name:      "obj1-name",
+							Namespace: "obj1-ns",
+							GroupKind: v1alpha1.GroupKind{
+								Group: "obj1-group",
+								Kind:  "obj1-kind",
+							},
+						},
+						{
+							Name:      "obj2-name",
+							Namespace: "obj2-ns",
+							GroupKind: v1alpha1.GroupKind{
+								Group: "obj2-group",
+								Kind:  "obj2-kind",
+							},
+						},
+					},
+				},
+				status: v1alpha1.ResourceGroupStatus{
+					ObservedGeneration: int64(invGeneration),
+					Conditions: []v1alpha1.Condition{
+						{
+							Type:   v1alpha1.Stalled,
+							Status: v1alpha1.FalseConditionStatus,
+						},
+					},
+					ResourceStatuses: []v1alpha1.ResourceStatus{
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj1-name",
+								Namespace: "obj1-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj1-group",
+									Kind:  "obj1-kind",
+								},
+							},
+							Status:    v1alpha1.Unknown,
+							Strategy:  v1alpha1.Apply,
+							Actuation: v1alpha1.ActuationSucceeded,
+							Reconcile: v1alpha1.ReconcileSucceeded,
+						},
+						{
+							ObjMetadata: v1alpha1.ObjMetadata{
+								Name:      "obj2-name",
+								Namespace: "obj2-ns",
+								GroupKind: v1alpha1.GroupKind{
+									Group: "obj2-group",
+									Kind:  "obj2-kind",
+								},
+							},
+							Status:    v1alpha1.Unknown,
+							Strategy:  v1alpha1.Apply,
+							Actuation: v1alpha1.ActuationPending,
+							Reconcile: v1alpha1.ReconcilePending,
+						},
+					},
+				},
+			}.build(t),
+		},
+	}
+	for name, tc := range testCases {
+		t.Run(name, func(t *testing.T) {
+			ic := NewInventoryConverter(declared.RootScope, invName, metadata.StatusEnabled)
+			rg, err := ic.InventoryToUnstructured(tc.fromObj, tc.toInv)
+			if tc.wantErr {
+				require.Error(t, err)
+				return
+			}
+			require.NoError(t, err)
+			require.Equal(t, invName, rg.GetName())
+			require.Equal(t, invNamespace, rg.GetNamespace())
+			// Set expected annotations
+			tc.wantObj.SetAnnotations(map[string]string{
+				metadata.ResourceManagementKey: metadata.ResourceManagementEnabled,
+				metadata.StatusModeKey:         metadata.StatusEnabled,
+			})
+			// Set expected labels
+			tc.wantObj.SetLabels(map[string]string{
+				common.InventoryLabel:       invID,
+				metadata.ManagedByKey:       metadata.ManagedByValue,
+				metadata.SyncNameLabel:      invName,
+				metadata.SyncNamespaceLabel: configsync.ControllerNamespace,
+				metadata.SyncKindLabel:      configsync.RootSyncKind,
+			})
+			if !equality.Semantic.DeepEqual(rg, tc.wantObj) {
```

Comment:
You can use testutil.AssertEqual to handle the comparison and diff printing for you. 

I don't think we actually want to use Semantic.DeepEqual in the test anyway. Would be good to validate it's exactly what we expect, just not similar, for catching minor regressions.

Reply:
done